### ISSUE TITLE
Custom saml_create_user value for different user models

### ISSUE THREAD

**Issue by linhmtran168** [apokalipto/devise_saml_authenticatable#203]
In our application, there are two different user systems, each with a different subdomain, all running in the same Rails application. One is for administrator so we want to set `saml_create_user` value to be true, one is for normal user so `saml_create_user`should be false (the user must be created in the system beforehand by the administrator).

Is there any way we can do that with with devise_saml_authenticatable?

**Comment by adamstegman** [comment]
Interesting idea! We have not set it up that way, but that's a great use case. I wonder if we'd want something in the model, like:

```ruby
devise :saml_authenticatable, :saml_creatable
```

Or if we'd want to configure that in `devise.rb`:

```ruby
config.saml_create_user = ->(model_class) { model_class == Admin }
```

**Comment by linhmtran168** [comment]
In my opinion, make `saml_create_user` and `saml_update_user` can accept a `Proc` may be a good idea.

**Comment by adamstegman** [comment]
I agree! We could even pass in the `auth_value` so that you can make finer-grained decisions based on the identifier. If you're interested in writing a PR, that could be very helpful!

**Comment by linhmtran168** [comment]
Yes, I will try to write a PR later when I have some free time.
